src/geo/projection/globe_util.js

```diff
@@ -655,6 +655,38 @@ export function isLngLatBehindGlobe(tr: Transform, lngLat: LngLat): boolean {
     return (globeTiltAtLngLat(tr, lngLat) > Math.PI / 2 * 1.01);
 }
 
+/**
+ * Check if poles are visible inside the current viewport
+ *
+ * @param {Transform} transform The current map transform.
+ * @returns {[boolean, boolean]} A tuple of booleans [northInViewport, southInViewport]
+ */
+export function polesInViewport(tr: Transform): [boolean, boolean] {
+    // Create matrix to screen coordinates
```

Comment:
IMO it's helpful to specify the clarify the space matrices transform both to and from.
```suggestion
    // Create matrix from ECEF to screen coordinates
```
Alternatively, maybe this could be a case for self-documenting code? e.g. `const ecefToScreenMatrix`